crates/control/src/role_assignment.rs

```diff
@@ -372,289 +285,320 @@ impl RoleAssignment {
+            Some(team_ball) if striker_trusts_team_ball(team_ball) => Role::Striker,
+            _ => Role::Loser,
         },
 
-        //Loser found ball and becomes Striker
-        (Role::Loser, Some(..), None) => (
-            Role::Striker,
-            true,
-            team_ball_from_seen_ball(detected_own_ball, current_pose, cycle_start_time),
+        (_other_role, _, Event::Striker(striker_event)) => claim_striker_or_other_role(
+            striker_event,
+            time_to_reach_kick_position,
+            player_number,
+            filtered_game_controller_state,
+            optional_roles,
         ),
 
-        // Edge-case, Loser found Ball at the same time as receiving a message
-        (Role::Loser, Some(..), Some(spl_message)) => match &spl_message.ball_position {
-            None => {
-                // another Striker became Loser, so we claim striker since we see a ball
-                (
-                    Role::Striker,
-                    true,
-                    team_ball_from_seen_ball(detected_own_ball, current_pose, cycle_start_time),
-                )
-            }
-            _ => decide_if_claiming_striker_or_other_role(
-                spl_message,
-                time_to_reach_kick_position,
-                player_number,
-                cycle_start_time,
-                filtered_game_controller_state,
-                optional_roles,
-            ),
-        },
-
-        //Searcher remains Searcher
-        (Role::Searcher, None, None) => (Role::Searcher, false, team_ball),
-
-        (Role::Searcher, None, Some(spl_message)) => match &spl_message.ball_position {
-            None => (Role::Searcher, false, team_ball), //edge-case, a striker (which should not exist) lost the ball
-            _ => decide_if_claiming_striker_or_other_role(
-                spl_message,
-                time_to_reach_kick_position,
-                player_number,
-                cycle_start_time,
-                filtered_game_controller_state,
-                optional_roles,
-            ),
+        // Striker remains Striker, sends message after timeout
+        (Role::Striker, true, Event::None) => Role::Striker,
+
+        // Edge-case, another Striker became Loser, so we claim striker since we see a ball
+        // TODO: On main, this sends a striker message immediately, ignoring the spl_striker_message_send_interval
+        //       but not the silence_interval_between_messages.
+        //       With the new implementation this is no longer possible since the message sending
+        //       is only based on the previous and new roles and does not consider the cause of the
+        //       transition.
+        //       This was already broken on main since the message would silently be dropped if the
+        //       edge case occured within the silence interval but might lead to a faster
+        //       striker convergence in this rare circumstance.
```

Comment:
marking this with a comment for discussion...

Reply:
Removed